src/components/BatchTransactionsSender/BatchTransactionsSender.ts

```diff
@@ -0,0 +1,157 @@
+import { useCallback, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'reduxStore/DappProviderContext';
+import {
+  accountSelector,
+  signedTransactionsSelector
+} from 'reduxStore/selectors';
+import { sendBatchTransactions } from 'services/transactions/sendBatchTransactions';
+import {
+  TransactionBatchStatusesEnum,
+  TransactionServerStatusesEnum
+} from 'types/enums.types';
+import { SignedTransactionsBodyType } from 'types/transactions.types';
+import {
+  clearAllTransactionsToSign,
+  setTxSubmittedModal,
+  setBatchTransactions,
+  updateSignedTransactions
+} from 'reduxStore/slices';
+import { setNonce } from 'utils/account/setNonce';
+import { safeRedirect } from 'utils/redirect';
+import { removeTransactionParamsFromUrl } from 'utils/transactions/removeTransactionParamsFromUrl';
+import { sequentialToFlatArray } from 'utils/transactions/batch/sequentialToFlatArray';
+
+/**
+ * Function used to redirect after sending because of Safari cancelling async requests on page change
+ */
+const optionalRedirect = (sessionInformation: SignedTransactionsBodyType) => {
+  const redirectRoute = sessionInformation.redirectRoute;
+  if (redirectRoute) {
+    safeRedirect(redirectRoute);
+  }
+};
+
+export const BatchTransactionsSender = () => {
+  const dispatch = useDispatch();
+
+  const { address, nonce } = useSelector(accountSelector);
+  const signedTransactions = useSelector(signedTransactionsSelector);
+
+  const sendingRef = useRef(false);
+
+  const clearSignInfo = () => {
+    dispatch(clearAllTransactionsToSign());
+    sendingRef.current = false;
+  };
+
+  const handleSendTransactions = useCallback(async () => {
+    const sessionIds = Object.keys(signedTransactions);
+
+    for (const sessionId of sessionIds) {
+      const sessionInformation = signedTransactions?.[sessionId];
```

Comment:
If `signedTransactions` can be undefined, `Object.keys(signedTransactions)` will throw error

Reply:
`signedTransactions` can't be undefined. Has an empty object as the default value. There is a wrong usage of `?` operator. Will be fixed